### DISCUSSION TITLE
What is the meanning for "schema", "meta-schema" and "consumer-schema" in plugins?

### DISCUSSION THREAD

**Discussion by cyflhn** [apache/apisix#6972] [General]
What are the meannings of "schema", "meta-schema" and "consumer-schema" in plugins? 
I did not find any description about these in documents.
I want to develop a customized plugin, but I don't know how to define the three schemas in my plugin.

**Comment by moonming** [comment]
@tzssangglass please take a look. Is APISIX has docs for these?

**Reply by tzssangglass** [reply]
yes, ref to: https://github.com/apache/apisix/blob/master/docs/en/latest/plugin-develop.md#schema-and-check

**Reply by cyflhn** [reply]
@tzssangglass I still did not find the meaning of those schema in the docs you provided.  The docs just shows how to define a shema.

**Reply by tzssangglass** [reply]
I think you need to understand what they mean first.

plugin-metadata: https://github.com/apache/apisix/blob/master/docs/en/latest/FAQ.md#what-is-the-difference-between-plugin-metadata-and-plugin-configs-in-apache-apisix

consumer: https://github.com/apache/apisix/blob/master/docs/en/latest/terminology/consumer.md
